misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
+        Node typeName = formatNode(defaultableParameterNode.typeName(), 1, 0);
+
+        if (defaultableParameterNode.paramName().isPresent()) {
+            Token paramName = formatToken(defaultableParameterNode.paramName().get(), 1, 0);
+            defaultableParameterNode = defaultableParameterNode.modify().withParamName(paramName).apply();
+        }
+
+        Token equalsToken = formatToken(defaultableParameterNode.equalsToken(), 1, 0);
+        Node expression = formatNode(defaultableParameterNode.expression(), this.trailingWS, this.trailingNL);
+
+        return defaultableParameterNode.modify()
+                .withAnnotations(annotations)
+                .withTypeName(typeName)
+                .withEqualsToken(equalsToken)
+                .withExpression(expression)
+                .apply();
+    }
+
+    @Override
+    public RestParameterNode transform(RestParameterNode restParameterNode) {
+
+        return super.transform(restParameterNode);
+    }
+
+    @Override
+    public SpreadFieldNode transform(SpreadFieldNode spreadFieldNode) {
+
+        return super.transform(spreadFieldNode);
+    }
+
+    @Override
+    public NamedArgumentNode transform(NamedArgumentNode namedArgumentNode) {
+
+        return super.transform(namedArgumentNode);
+    }
+
+    @Override
+    public RestArgumentNode transform(RestArgumentNode restArgumentNode) {
+
+        return super.transform(restArgumentNode);
+    }
+
+    @Override
+    public ObjectTypeDescriptorNode transform(ObjectTypeDescriptorNode objectTypeDescriptorNode) {
+
+        return super.transform(objectTypeDescriptorNode);
+    }
+
+    @Override
+    public ObjectConstructorExpressionNode transform(ObjectConstructorExpressionNode objectConstructorExpressionNode) {
+
+        return super.transform(objectConstructorExpressionNode);
+    }
+
+    @Override
+    public ObjectFieldNode transform(ObjectFieldNode objectFieldNode) {
+
+        return super.transform(objectFieldNode);
+    }
+
+    @Override
+    public RecordRestDescriptorNode transform(RecordRestDescriptorNode recordRestDescriptorNode) {
+
+        return super.transform(recordRestDescriptorNode);
+    }
+
+    @Override
+    public AnnotationDeclarationNode transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        if (annotationDeclarationNode.metadata().isPresent()) {
+            MetadataNode metadata = formatNode(annotationDeclarationNode.metadata().get(), 0, 1);
+            annotationDeclarationNode = annotationDeclarationNode.modify().withMetadata(metadata).apply();
+        }
+
+        Token visibilityQualifier = formatToken(annotationDeclarationNode.visibilityQualifier(), 1, 0);
+        Token constKeyword = formatToken(annotationDeclarationNode.constKeyword(), 1, 0);
+        Token annotationKeyword = formatToken(annotationDeclarationNode.annotationKeyword(), 1, 0);
+        Node typeDescriptor = formatNode(annotationDeclarationNode.typeDescriptor(), 1, 0);
```

Comment:
This also seems to be a bug in syntax-tree. Both `constKeyword` and `typeDescriptor` needs to be optional. Lets create an issue

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/26328